src/Components/Articles.jsx

```diff
@@ -6,24 +6,33 @@ import ArticleList from "./ArticlesList.jsx";
 function Articles() {
   const [articles, setArticles] = useState();
   const [selectedTopic, setSelectedTopic] = useState(" ");
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     if (selectedTopic) {
       fetchArticles().then((reponse) => {
         setArticles(reponse.data.articles);
+        setLoading(false);
       });
     } else {
       fetchArticles(selectedTopic).then((response) => {
         setArticles(response.data.articles);
+        setLoading(false);
       });
     }
   }, [selectedTopic]);
 
```

Comment:
You're definitely on the right track here, but if you route by :topic then you can use useParams to grab if there's a topic or not and render this component on multiple routes. Then, you can just call fetchArticles once without the if condition, always passing it a topic. In your axios call you can add a second argument for params, there's an example in the notes. I think this is working anyway but could be tidied up a little. I'll be doing something like this in the seminar tomorrow!

Reply:
Ooh, nice will look forward to that seminar to get further understanding!  Thanks for the feedback Christian 😊

Reply:
That seminar was really useful and makes the comment a lot clearer to me now 😊